client.go

```diff
@@ -72,9 +72,7 @@ type Marathon interface {
 	// get a list of tasks for a specific application
 	Tasks(application string) (*Tasks, error)
 	// get a list of all tasks
-	AllTasks() (*Tasks, error)
-	// get a listing of the task ids
-	ListTasks() ([]string, error)
+	AllTasks(status string) (*Tasks, error)
```

Comment:
I think I would rather help a user by explicitly providing possible parameters. In this case you don't need to look to the api and you have compile time check that your params are right.
There are two downsides with this approach:
1) as it was mentioned: api might change
2) providing nil values will be ugly

We can solve `2)` by wrapping request params into structs, and I think it might look actually nice.

I think at this point I will do `url.Values` and then we can decide if we want to refactor the way parameters are passed.